session/manager.go

```diff
@@ -56,7 +57,7 @@ type publisher interface {
 
 // ConfigProvider is able to handle config negotiations
 type ConfigProvider interface {
-	ProvideConfig(sessionID string, sessionConfig json.RawMessage) (*ConfigParams, error)
+	ProvideConfig(sessionID string, sessionConfig json.RawMessage, conn *net.UDPConn) (*ConfigParams, error)
```

Comment:
i would not inject conn here, would pass just whats required: 
i.e.: `remoteConn.LocalAddr().(*net.UDPAddr).Port`
We should handle streams separately, not in config provider.

Reply:
We cannot just send LocalPort, it depends on transport what to do with a connection.
OpenVPN needs to start proxy to connect multiple clients to the single OpenVPN process, so it reuses that connection.

Wireguard is closing the connection and starts its own handlers on that ports. We even can use both LocalPort and RemotePort to make the connection more secure.

We never know what will be next transport and how it will need to handle the provided connection.

Reply:
Ok, then renaming `ProvideConfig` to more sensible naming would do for some time :)